app/models/concerns/category_hashtag.rb

```diff
@@ -51,15 +50,15 @@ def query_from_cached_categories(category_slugs, cached_categories)
           # by its slug then find the child by its slug and its parent's
           # ID to make sure they match.
           if child_slug.present?
-            parent_category = cached_categories.find { |cat| cat[:slug].downcase == parent_slug }
+            parent_category = categories.find { |cat| cat.slug.downcase == parent_slug }
```

Comment:
Could potentially replace instances of `#downcase` + `#==` with [`#casecmp?`](https://ruby-doc.org/core-3.1.1/String.html#method-i-casecmp-3F) which is about twice as fast as it's one less method call and no new object allocations. However in the context of what this does, that's probably not even measurable. Leaving the comment anyway for people to learn about the method. 🙂

Reply:
Nice, I do forget about this, will change it over :+1: